### ISSUE TITLE
[RFE] Provide "all applications" to rules creation

### ISSUE THREAD

**Issue by mattiaverga** [fedora-infra/fmn#876]
On creating a new rule (for example for "My Events" or "My artifacts") I would like to select "all applications", so in future when new apps are added I don't have to remember to update my rule.

**Comment by abompard** [comment]
Ah, if you don't select any applications, it will disable the filter and do what you want. That's probably not clear enough.

**Comment by Precious000** [comment]
Hi there! Can I get assigned this task? I’ll like to work on it.

**Comment by abompard** [comment]
@Precious000 If my records are correct, you have been assigned to #987. Would you prefer to work on this one?

**Comment by Precious000** [comment]
@abompard Yes please. Thank you!

**Comment by Precious000** [comment]
Better still, I’ll maintain the previous one, thanks a bunch boss!

**Comment by RuthM-K** [comment]
@abompard can I get assigned this task?

**Comment by abompard** [comment]
@RuthM-K done, have at it! :-)

**Comment by RuthM-K** [comment]
Thank you, On it!

**Comment by RuthM-K** [comment]
@abompard When I cloned the repo, some modules were not found
How do I resolve the errors?
Again, the application does not run when I npm run dev. npm install is not working as well
![fedora](https://github.com/fedora-infra/fmn/assets/144100504/b854c59a-0d27-4d9f-8c95-11664a73ece3)
Not sure if the attached shows the image

**Comment by abompard** [comment]
What does `npm install` say?

**Comment by RuthM-K** [comment]
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /home/ruth/Fedora/fmn/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/home/ruth/Fedora/fmn/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in: /home/ruth/.npm/_logs/2024-03-19T16_07_23_694Z-debug-0.log

That is the error

**Comment by RuthM-K** [comment]
Any guidance? @abompard

**Comment by abompard** [comment]
Yeah you need to run the `npm install` in the "frontend" directory.

**Comment by sgollor** [comment]
Greetings, Sam here. I am interested in helping with this issue. Let's go! 💪

**Comment by sgollor** [comment]
I've made changes to two(2) files `BaseFilterApplication.vue` and `FilterApplication.vue`

It adds helper text to clarify that leaving the Applications field blank/unselected means your rule applies to every application, including future ones.

I am not currently able to fully create a rule because I need other Fedora systems running locally.

@abompard @mattiaverga What do you think? Is there more to add?
